sdcm/nemesis.py

```diff
@@ -4417,6 +4423,106 @@ def disrupt_add_remove_mv(self):
+            verify audit log content,
+            reduce categories by excluding DML and QUERY,
+            verify DDL are logged in audit log correctly. Leaves audit log enabled this way.
+        """
+        if not self.target_node.is_enterprise:
+            raise UnsupportedNemesis("Auditing feature is only supported by Scylla Enterprise")
+
+        if store == "syslog" and self._is_it_on_kubernetes():
+            raise UnsupportedNemesis("syslog store is not supported on Kubernetes #scylla-operator/1299")
+
+        audit = Audit(self.cluster)
+
+        if audit.is_enabled():
+            audit.disable()
+            raise UnsupportedNemesis("Audit was enabled -> disabling it")
+
+        audit_keyspace = "audit_keyspace"
+        keyspaces_for_audit = [audit_keyspace]
+        InfoEvent(f"Enabling full audit for keyspaces: {keyspaces_for_audit}").publish()
+        audit_config = AuditConfiguration(
+            store=store,
+            categories=["DCL", "DDL", "AUTH", "ADMIN", "DML", "QUERY"],
+            keyspaces=keyspaces_for_audit,
+            tables=[],
+        )
+        try:
+            audit.configure(audit_config)
+            keyspace_name = keyspaces_for_audit[0]
+            errors = []
+            audit_start = datetime.datetime.now()
+            InfoEvent(message='Writing/Reading data from audited keyspace').publish()
+            write_cmd = f"cassandra-stress write no-warmup cl=ALL n=1000 -schema" \
+                        f" 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)" \
+                        f" keyspace={audit_keyspace}' -mode cql3 native -rate 'threads=1 throttle=1000/s'" \
+                        f" -pop seq=1..1000 -col 'n=FIXED(1) size=FIXED(128)' -log interval=5"
+            write_thread = self.tester.run_stress_thread(
+                stress_cmd=write_cmd, round_robin=True, stop_test_on_failure=False)
+            self.tester.verify_stress_thread(cs_thread_pool=write_thread)
+            read_cmd = f"cassandra-stress read no-warmup cl=ALL n=1000 " \
+                       f" -schema 'replication(strategy=NetworkTopologyStrategy,replication_factor=3)" \
+                       f" keyspace={audit_keyspace}' -mode cql3 native -rate 'threads=1 throttle=1000/s'" \
+                       f" -pop seq=1..1000 -col 'n=FIXED(1) size=FIXED(128)' -log interval=5"
+            read_thread = self.tester.run_stress_thread(
+                stress_cmd=read_cmd, round_robin=True, stop_test_on_failure=False)
+            self.tester.verify_stress_thread(cs_thread_pool=read_thread)
+            InfoEvent(message='Verifying Audit table contents').publish()
+            rows = audit.get_audit_log(from_datetime=audit_start, category="DML", limit_rows=1100)
+            # filter out USE keyspace rows due to https://github.com/scylladb/scylla-enterprise/issues/3169
+            rows = [row for row in rows if not row.operation.startswith("USE")]
```

Comment:
If we expect only DML, perhaps we should filter out every other category, not only the USE KS?

Reply:
`USE KS` is DML category, that's why it is filtered later manually like this